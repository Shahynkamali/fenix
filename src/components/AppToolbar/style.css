.toolbar {
  @apply bg-white w-full;
}

.toolbar__container {
  @apply relative flex min-w-5 p-3 shadow min-h-3;
}

.toolbar__button-container {
  @apply hidden;
}

.toolbar__content {
  @apply flex w-full;
}

.toolbar__content--horizontal {
  @apply flex-wrap justify-evenly md:justify-start;
}

.toolbar__content--vertical {
  @apply flex-col items-center justify-start shadow px-8 pt-2 pb-1;
}

.toolbar__button {
  @apply inline-flex items-center justify-center p-2 rounded-md text-gray-400 transition duration-150 ease-in-out;
}

.toolbar__button:hover {
  @apply text-gray-500 bg-gray-100;
}

.toolbar__button:focus {
  @apply outline-none bg-gray-100;
}

.toolbar__icon {
  @apply block h-6 w-6;
}

@media (max-width: 540px) {
  .toolbar__content--horizontal {
    @apply hidden;
  }

  .toolbar__button-container {
    @apply block absolute left-2/4 top-2/4 ;

    -webkit-transform: translateX(-50%);
    transform: translateX(-50%) translateY(-50%);
  }
}
