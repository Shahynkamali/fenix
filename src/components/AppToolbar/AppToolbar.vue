<template>
  <nav
    class="toolbar"
  >
    <div
      class="toolbar__container"
      :class="customStyles"
    >
      <div
        class="toolbar__button-container"
      >
        <button
          class="toolbar__button"
          aria-label="Toolbar menu"
          aria-expanded="false"
          @click="toggleToolBarButton"
        >
          <HamburgerSVG
            v-if="!isToolBarOpen"
            aria-label="Toolbar menu open"
            class="toolbar__icon"
          />
          <CloseSVG
            v-else
            aria-label="Toolbar menu close"
            class="toolbar__icon"
          />
        </button>
      </div>
      <div class="toolbar__content toolbar__content--horizontal">
        <slot name="toolBarContent" />
      </div>
    </div>
    <transition name="slide">
      <div
        v-show="isToolBarOpen"
        class="toolbar__content toolbar__content--vertical"
      >
        <slot name="toolBarContent" />
      </div>
    </transition>
  </nav>
</template>

<script lang="ts">
import {
  Component, Vue, Watch, Prop,
} from 'vue-property-decorator';
import CloseSVG from '@/assets/svgs/close.svg';
import HamburgerSVG from '@/assets/svgs/hamburger.svg';
import './style.css';

@Component({
  name: 'AppToolbar',
  components: {
    CloseSVG,
    HamburgerSVG,
  },
})
export default class AppToolbar extends Vue {
  @Prop({ required: false })
    readonly customStyles!: string;

  @Watch('viewPortWidth')
  watchViewPortChange(viewPortWidth: number) {
    if (viewPortWidth <= 539 && this.isToolBarOpen) {
      this.isToolBarOpen = false;
    }
  }

  viewPortWidth = 0;

  isToolBarOpen = false;

  private toggleToolBarButton(): void {
    this.isToolBarOpen = !this.isToolBarOpen;
  }

  private handleResize(): void {
    this.viewPortWidth = window.innerWidth;
  }

  created(): void {
    window.addEventListener('resize', this.handleResize);
    this.handleResize();
  }

  destroyed(): void {
    window.removeEventListener('resize', this.handleResize);
  }
}
</script>

<style scoped>
.slide-enter-active {
  -moz-transition-duration: 0.3s;
  -webkit-transition-duration: 0.3s;
  -o-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -moz-transition-timing-function: ease-in;
  -webkit-transition-timing-function: ease-in;
  -o-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.slide-leave-active {
  -moz-transition-duration: 0.3s;
  -webkit-transition-duration: 0.3s;
  -o-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -moz-transition-timing-function: cubic-bezier(0, 1, 0.5, 1);
  -webkit-transition-timing-function: cubic-bezier(0, 1, 0.5, 1);
  -o-transition-timing-function: cubic-bezier(0, 1, 0.5, 1);
  transition-timing-function: cubic-bezier(0, 1, 0.5, 1);
}

.slide-enter-to,
.slide-leave {
  max-height: 100px;
  overflow: hidden;
}

.slide-enter,
.slide-leave-to {
  overflow: hidden;
  max-height: 0;
}
</style>
