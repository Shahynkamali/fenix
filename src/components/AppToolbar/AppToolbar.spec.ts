import { mount } from '@vue/test-utils';
import UserSVG from '@/assets/svgs/users.svg';
import InformationCircleSVG from '@/assets/svgs/information-circle.svg';
import CloseSVG from '@/assets/svgs/close.svg';
import HamburgerSVG from '@/assets/svgs/hamburger.svg';
import AppToolbar from './AppToolbar.vue';

describe('AppToolbar', () => {
  const factory = (slotsProp?: any, propsData?: any) => mount(AppToolbar, {
    propsData: {
      ...propsData,
    },
    slots: {
      ...slotsProp,
    },
    stubs: {
      UserSVG,
      InformationCircleSVG,
      CloseSVG,
      HamburgerSVG,
    },
  });
  it('should render the template injected in the slot', () => {
    const slotToolBarContent = {
      toolBarContent: `
        <div class="stub-svg-container">
          <UserSVG/>
        </div>
        <div class="stub-svg-container">
          <InformationCircleSVG/>
        </div>
      `,
    };
    const wrapper = factory(slotToolBarContent);
    expect(wrapper.findAll('.stub-svg-container').length).toBe(4);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should set the state of isToolBarOpen to true on the click of the button', () => {
    global.innerWidth = 330;
    global.dispatchEvent(new Event('resize'));
    const wrapper = factory();
    wrapper.find('button').trigger('click');
    expect(wrapper.vm.$data.isToolBarOpen).toBeTruthy();
    expect(wrapper.findComponent(HamburgerSVG).exists()).toBe(true);
    expect(wrapper.element).toMatchSnapshot();
  });
});
