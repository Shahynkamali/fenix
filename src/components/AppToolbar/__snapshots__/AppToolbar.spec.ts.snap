// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppToolbar should render the template injected in the slot 1`] = `
<nav
  class="toolbar"
>
  <div
    class="toolbar__container"
  >
    <div
      class="toolbar__button-container"
    >
      <button
        aria-expanded="false"
        aria-label="Toolbar menu"
        class="toolbar__button"
      >
        <svg
          aria-label="Toolbar menu open"
          class="toolbar__icon"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M4 6h16M4 12h16M4 18h16"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
     
    <div
      class="toolbar__content toolbar__content--horizontal"
    >
      <div
        class="stub-svg-container"
      >
        <svg
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 4.354a4 4 0 110 5.292M15 21H3v-1a6 6 0 0112 0v1zm0 0h6v-1a6 6 0 00-9-5.197M13 7a4 4 0 11-8 0 4 4 0 018 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
       
      <div
        class="stub-svg-container"
      >
        <svg
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
  </div>
   
  <transition-stub
    name="slide"
  >
    <div
      class="toolbar__content toolbar__content--vertical"
      style="display: none;"
    >
      <div
        class="stub-svg-container"
      >
        <svg
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 4.354a4 4 0 110 5.292M15 21H3v-1a6 6 0 0112 0v1zm0 0h6v-1a6 6 0 00-9-5.197M13 7a4 4 0 11-8 0 4 4 0 018 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
       
      <div
        class="stub-svg-container"
      >
        <svg
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
  </transition-stub>
</nav>
`;

exports[`AppToolbar should set the state of isToolBarOpen to true on the click of the button 1`] = `
<nav
  class="toolbar"
>
  <div
    class="toolbar__container"
  >
    <div
      class="toolbar__button-container"
    >
      <button
        aria-expanded="false"
        aria-label="Toolbar menu"
        class="toolbar__button"
      >
        <svg
          aria-label="Toolbar menu open"
          class="toolbar__icon"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M4 6h16M4 12h16M4 18h16"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
     
    <div
      class="toolbar__content toolbar__content--horizontal"
    />
  </div>
   
  <transition-stub
    name="slide"
  >
    <div
      class="toolbar__content toolbar__content--vertical"
      style="display: none;"
    />
  </transition-stub>
</nav>
`;
